test/index.js

```diff
@@ -249,7 +253,6 @@ describe('Generic', function () {
       Object.prototype.toString = originalObjectToString; // eslint-disable-line no-extend-native
```

Comment:
I have a feeling you'll need to change the sham behaviour to test for getter properties, right now it only asserts that the `[Symbol.toStringTag]` prop is a function (which in the case of your class, it isn't)